File: eco-core/core-backend/src/main/java/com/willfp/eco/internal/factory/EcoNamespacedKeyFactory.java
package com.willfp.eco.internal.factory;

import com.willfp.eco.core.PluginDependent;
import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.factory.NamespacedKeyFactory;
import org.bukkit.NamespacedKey;
import org.jetbrains.annotations.NotNull;

public class EcoNamespacedKeyFactory extends PluginDependent<EcoPlugin> implements NamespacedKeyFactory {
    public EcoNamespacedKeyFactory(@NotNull final EcoPlugin plugin) {
        super(plugin);
    }

    @Override
    public NamespacedKey create(@NotNull final String key) {
        return new NamespacedKey(this.getPlugin(), key);
    }
}

File: eco-core/core-backend/src/main/kotlin/com/willfp/eco/internal/factory/EcoNamespacedKeyFactory.kt
package com.willfp.eco.internal.factory

import com.willfp.eco.core.EcoPlugin
import com.willfp.eco.core.factory.NamespacedKeyFactory
import org.bukkit.NamespacedKey

class EcoNamespacedKeyFactory(private val plugin: EcoPlugin) : NamespacedKeyFactory {
    override fun create(key: String): NamespacedKey {
        return NamespacedKey(plugin.id, key)
    }
}
